lib/grape/middleware/versioner.rb

```diff
@@ -13,21 +13,12 @@ module Middleware
     module Versioner
       module_function
 
-      # @param strategy [Symbol] :path, :header or :param
+      # @param strategy [Symbol] :path, :header, :accept_version_header or :param
       # @return a middleware class based on strategy
       def using(strategy)
-        case strategy
-        when :path
-          Path
-        when :header
-          Header
-        when :param
-          Param
-        when :accept_version_header
-          AcceptVersionHeader
-        else
-          raise Grape::Exceptions::InvalidVersionerOption.new(strategy)
-        end
+        Grape::Middleware::Versioner.const_get(:"#{strategy.to_s.classify}")
+      rescue NameError
```

Comment:
I don't like that we rely on an exception here. It also allows you to register a strategy without anyone noticing, which ends up being used as an API.

Reply:
I do understand your point about registering without anyone noticing but on the other hand we're already embracing the pattern through [coercing](https://github.com/ruby-grape/grape/blob/2b8567a1ca49b3d0236fa82deb2edef34c8839cf/lib/grape/validations/types/primitive_coercer.rb#L35) and [validators](https://github.com/ruby-grape/grape/blob/2b8567a1ca49b3d0236fa82deb2edef34c8839cf/lib/grape/validations.rb#L25). Now, everyone can build their own `versioner` and they just need to follow the naming convention like the validators.

Reply:
In the future we could imagine having a separate gem just for versioning. It's a bit like how [Faraday](https://github.com/lostisland/faraday/blob/main/UPGRADING.md#adapters-have-moved) evolved through time by moving out the adapters. I know versioning ain't that big compared to adapters but its the idea

Reply:
Ok, I buy the argument of we're already doing it.